lib/geminabox/gem_list_merge.rb

```diff
@@ -1,23 +1,38 @@
+require "set"
+
 module Geminabox
   class GemListMerge
     attr_accessor :list
 
     IGNORE_DEPENDENCIES = 0..-2
 
-    def self.from(*lists)
-      lists.map{|list| new(list)}.inject(:merge)
+    def self.merge(strategy, lists)
```

Comment:
WDYT about allowing `strategy` to default to `nil` and use the existing merge strategy `if strategy.nil?` in order to allow for a non-breaking approach here?

Reply:
Started with that. I think the vulnerability is serious enough that it merits breaking, I'm afraid. This needs to become the new default rather than requiring opt-in.